src/Control/Effect/Diagnostics.hs

```diff
@@ -89,6 +92,12 @@ fatal = send . Fatal
 fatalText :: Has Diagnostics sig m => Text -> m a
 fatalText = fatal
 
+-- | Throw a generic error message on IO error
+fatalOnIOException :: (Has (Lift IO) sig m, Has Diagnostics sig m) => m a -> m a
+fatalOnIOException go = catch go die'
+  where
+    die' (e :: IOException) = fatalText ("io exception: " <> toText (show e))
```

Comment:
I think this might be a bit of a debugging footgun, since it's easy to get a diagnostic error with no real context for what caused it.

What if we required using `context` to wrap this catching function?

`fatalOnIOException msg go = context msg $ catch go die'`

Reply:
I like it! Done.